src/OrbitGl/CallstackTooltipUtils.cpp

```diff
@@ -0,0 +1,169 @@
+  ORBIT_CHECK(bottom_line_count < max_lines);
+
+  size_t callstack_size = callstack.frames().size();
+  const size_t bottom_n = std::min(bottom_line_count, callstack_size);
+  const size_t top_n = std::min(max_lines, callstack_size) - bottom_n;
+
+  constexpr size_t kShortenedForReadabilityFakeIndex = std::numeric_limits<size_t>::max();
+  constexpr const char* kShortenedForReadabilityString = "<i>... shortened for readability ...</i>";
+  std::vector<size_t> frame_indices_to_display;
+  frame_indices_to_display.reserve(top_n);
+  for (size_t i = 0; i < top_n; ++i) {
+    frame_indices_to_display.push_back(i);
+  }
+  if (callstack_size > max_lines) {
+    frame_indices_to_display.push_back(kShortenedForReadabilityFakeIndex);
+  }
+  for (size_t i = callstack_size - bottom_n; i < callstack_size; ++i) {
+    frame_indices_to_display.push_back(i);
+  }
+
+  std::string result;
+  for (size_t frame_index : frame_indices_to_display) {
+    if (frame_index == kShortenedForReadabilityFakeIndex) {
+      result.append(kShortenedForReadabilityString).append("<br/>");
+      continue;
+    }
+
+    static const std::string kModuleFunctionSeparator{" | "};
+    const UnformattedModuleAndFunctionName module_and_function_name =
+        SafeGetModuleAndFunctionName(callstack, frame_index, capture_data, module_manager);
+    const std::string formatted_module_name = FormatModuleName(module_and_function_name);
+    const std::string formatted_function_name = FormatFunctionName(
+        module_and_function_name, max_line_length - module_and_function_name.module_name.size() -
+                                      kModuleFunctionSeparator.size());
+    const std::string formatted_module_and_function_name =
+        absl::StrCat(formatted_module_name, kModuleFunctionSeparator, formatted_function_name);
+    // The first frame is always correct.
+    if (callstack.IsUnwindingError() && frame_index > 0) {
+      result += absl::StrFormat("<span style=\"color:%s;\">%s</span><br/>", kUnwindErrorColorString,
+                                formatted_module_and_function_name);
+    } else {
+      result.append(formatted_module_and_function_name).append("<br/>");
+    }
+  }
+
+  return result;
+}
+
+FormattedModuleAndFunctionName FormatInnermostFrameOfCallstackForTooltip(
+    const orbit_client_data::CallstackInfo& callstack,
+    const orbit_client_data::CaptureData& capture_data,
+    const orbit_client_data::ModuleManager& module_manager) {
+  UnformattedModuleAndFunctionName innermost_module_and_function_name =
+      SafeGetModuleAndFunctionName(callstack, 0, capture_data, module_manager);
+  std::string innermost_formatted_function_name =
+      FormatFunctionName(innermost_module_and_function_name, std::numeric_limits<size_t>::max());
+  std::string innermost_formatted_module_name =
+      FormatModuleName(innermost_module_and_function_name);
+
+  return {
+      innermost_formatted_module_name,
+      innermost_formatted_function_name,
+  };
```

Comment:
Is the trailing comma intended or...?
```suggestion
  return {innermost_formatted_module_name, innermost_formatted_function_name};
```

Reply:
Done.